### PR TITLE
223 openalex harvester

### PR DESCRIPTION
**Pull request by Paldana99**

### CODE REVIEW

File: app/harvesters/abstract_document_type_converter.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    HARVESTER: str
+
+    TYPES_MAPPING: dict[str, tuple[str, str]]
+
+    UNKNOWN_CODE = ("http://data.crisalid.org/ref/document_types/unknown", "Unknown")
+
+    def __init__(self) -> None:
+        self.harvester = self.HARVESTER
+        self.types_mapping = self.TYPES_MAPPING
```
Comment (JulienGaladrim) [inline app/harvesters/abstract_document_type_converter.py:19]: I think I wouldn't have defined constants here. Also I'd let the necessary constant outside the class. I would have written it like this :  

```
UNKNOWN_CODE = ("http://data.crisalid.org/ref/document_types/unknown", "Unknown")

class AbstractDocumentTypeConverter(ABC):

    harvester: str

    types_mapping: dict[str, tuple[str, str]]
```

File: app/harvesters/hal/hal_document_type_converter.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
-from loguru import logger
+from app.harvesters.abstract_document_type_converter import (
+    AbstractDocumentTypeConverter,
+)
 
 
-class HalDocumentTypeConverter:
+class HalDocumentTypeConverter(AbstractDocumentTypeConverter):
```
Comment (JulienGaladrim) [inline app/harvesters/hal/hal_document_type_converter.py:6]: Here I'd also let the constants outside and set the class attributes values like this :

```
HARVESTER = "HAL"
TYPES_MAPPING = [...]

class HalDocumentTypeConverter(AbstractDocumentTypeConverter):
    
    this.harvester = HARVESTER
    
    this.types_mapping = TYPES_MAPPING

File: app/harvesters/open_alex/open_alex_references_converter.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        """
+        Convert raw data from OpenAlex to a normalised Reference object
+        :param raw_data: raw data from OpenAlex
+        :return: Reference object
+        """
+        json_payload = raw_data.payload
+        new_ref = Reference()
+
+        language = json_payload.get("language", None)
+        new_ref.titles.append(self._title(json_payload, language))
```
Comment (jdp1ps) [inline app/harvesters/open_alex/open_alex_references_converter.py:34]: With ORCID 0000-0002-8156-2229 : 

```
Task exception was never retrieved
future: <Task finished name='openalex_harvester_retrieval_2' coro=<AbstractHarvester.run() done, defined at svp-harvester/app/harvesters/abstract_harvester.py:90> exception=IntegrityError('(sqlalchemy.dialects.postgresql.asyncpg.IntegrityError) <class \'asyncpg.exceptions.NotNullViolationError\'>: null value in column "value" of relation "reference_literal_fields" violates not-null constraint\nDETAIL:  Failing row contains (title, 188, null, en, 91).')>
...
    raise translated_error from error
sqlalchemy.dialects.postgresql.asyncpg.AsyncAdapt_asyncpg_dbapi.IntegrityError: <class 'asyncpg.exceptions.NotNullViolationError'>: null value in column "value" of relation "reference_literal_fields" violates not-null constraint
DETAIL:  Failing row contains (title, 188, null, en, 91).
```

Comment (jdp1ps) [inline app/harvesters/open_alex/open_alex_references_converter.py:34]: Same error with ORCID 0000-0001-9603-7681

**Review by jdp1ps — CHANGES_REQUESTED** [review]
Error with ORCID

File: app/harvesters/open_alex/open_alex_references_converter.py (hunk 3 above)
Comment (Paldana99) [reply app/harvesters/open_alex/open_alex_references_converter.py:34]: It was due a document with title None. In that case, it save in DB an empty string.

Comment (Paldana99) [reply app/harvesters/open_alex/open_alex_references_converter.py:34]: One of his document: https://openalex.org/works/W2069786235 has an asbtract of lenght 2705. Causing: 
> HINT:  Values larger than 1/3 of a buffer page cannot be indexed.

Problem is discussed in #230

**Review by jdp1ps — APPROVED** [review]

**Pull request merged**